### PR TITLE
Added Analytics service

### PR DESCRIPTION
**Pull request by eladBenedict**

### CODE REVIEW

File: api_v3/lib/KalturaErrors.php [hunk 1]
```diff
@@ -582,6 +582,12 @@ class KalturaErrors extends APIErrors
 	 * Live reports Service
 	 */
 	const LIVE_REPORTS_WS_FAILURE = 'LIVE_REPORTS_WS_FAILURE;;failed to retrieve live analytics';
+	
+	/*
+	 * Analytics Service
+	 */
+	const ANALYTICS_QUERY_FAILURE = 'REPORTS_QUERY_FAILURE;ERROR_MSG;failed to retrieve analytics data - [@ERROR_MSG@]';
```
Comment (tan-tan-kanarek) [inline api_v3/lib/KalturaErrors.php]: The attributes in the message created to enable future multilinguality, have attribute that contains description kind of missing the point.
Would it be possible to reflect all the different errors in the kAnalony as different errors in the API?

File: api_v3/lib/types/report/KalturaReportFilter.php [hunk 2]
```diff
@@ -0,0 +1,18 @@
+<?php
+/**
+ * @package api
+ * @subpackage objects
+ */
+class KalturaReportFilter extends KalturaObject
+{
+	/**
+	 * @var string
```
Comment (tan-tan-kanarek) [inline api_v3/lib/types/report/KalturaReportFilter.php]: Please add description to define the content and structure of this string, it's needed for the auto-generated documentation.

File: api_v3/services/AnalyticsService.php [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+	 * @param int $to query end time (in unix time)
+	 * @param string $metrics comma separated metrics list
+	 * @param string $dimensions comma separated dimensions list
+	 * @param KalturaReportFilterArray $filters array of filters
+	 * @return KalturaReportResponse
+	 */
+	public function queryAction($from, $to, $metrics, $dimensions = null, $filters = null)
+	{
+		$dimensionsArr = $this->extractDimensions($dimensions);
+		KalturaLog::info('analytics query - extracted dimensions: ' . var_export($dimensionsArr, true));
```
Comment (tan-tan-kanarek) [inline api_v3/services/AnalyticsService.php]: For all log messages, please remove "analytics query" prefix, the logs already contain the class and the method.
Also, consider to remove unneeded logs, and change those you keep to DEBUG.

File: api_v3/services/AnalyticsService.php [hunk 4]
```diff
@@ -0,0 +1,175 @@
...
+ * @package api
+ * @subpackage services
+ */
+class AnalyticsService extends KalturaBaseService
+{
+	/**
+	 * report query action allows to get a analytics data for specific query dimensions, metrics and filters.
+	 *
+	 * @action query
+	 * @param int $from query start time (in unix time)
```
Comment (tan-tan-kanarek) [inline api_v3/services/AnalyticsService.php]: Change int to time.

File: api_v3/services/AnalyticsService.php [hunk 5]
```diff
@@ -0,0 +1,175 @@
...
+	 * @param int $to query end time (in unix time)
+	 * @param string $metrics comma separated metrics list
+	 * @param string $dimensions comma separated dimensions list
+	 * @param KalturaReportFilterArray $filters array of filters
+	 * @return KalturaReportResponse
+	 */
+	public function queryAction($from, $to, $metrics, $dimensions = null, $filters = null)
```
Comment (tan-tan-kanarek) [inline api_v3/services/AnalyticsService.php]: Use one object that contains all the attributes, we may need to add more attributes in the future.

File: api_v3/services/AnalyticsService.php [hunk 6]
```diff
@@ -0,0 +1,175 @@
...
+	{
+		$res = array();
+		if ($filters != null)
+		{
+			$res = array_map(array($this, 'extractFilter'), $filters->toArray());
+		}
+
+		// Add a filter for the current partner
+		$partnerFilter = array();
+		$partnerFilter['dimension'] = "partner";
```
Comment (tan-tan-kanarek) [inline api_v3/services/AnalyticsService.php]: Please use constant, enum if possible.

File: api_v3/lib/KalturaErrors.php (hunk 1 above)
Comment (eladBenedict) [inline api_v3/lib/KalturaErrors.php]: Good point. Yes, it is certainly possible. I'll take it up with @kerkero to see how this fits in the work plan.

File: api_v3/services/AnalyticsService.php (hunk 5 above)
Comment (eladBenedict) [inline api_v3/services/AnalyticsService.php]: This form is consistent with the rest of the API. Additionally I don't see the added value in creating such an artificial construct for a top level method

**Pull request closed without merging**